### PR TITLE
#73/conversation container

### LINKED ISSUE #73: Fix viewport height to account for nav and footer 🍌
The message form should remain visible when scrolling, this means adjusting the viewport height to account for the navbar and bottom bar

### PR DESCRIPTION
**Pull request by nichgalzin**
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review on my code
- [x] I have made corresponding changes to the documentation (if any were needed)
- [x] My changes generate no new warnings or errors
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and previously existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

# Description

**Closes #73 **  

- This PR Fixes the displayed view when the individual view is displayed on mobile screen and attempts to begin fixing styling issue on devices of different sizes:

### Files changed

app/(dashboard)/conversations/page.tsx - moves context to the layout for the dashboard route.
components/Footer.tsx - hides mobile navbar in message view
components/Header.tsx - updates content based on view and route
components/filter/BooksSubcategory.tsx - type fix
components/filter/QuickBrowse.tsx- typo fix
components/icons/BackArrowIcon.tsx - new reusable component
components/layout/FullHeightComponent.tsx - Layout component to dynamically calculation the height of parent 
      components.
components/messaging/ConversationWrapper.tsx - back button moved to header
components/messaging/CurrentConversation.tsx - conversation partner moved to header
context/Providers.tsx - new wrapper client component to hold context providers

Include one to two lines on the necessary changes to each file.

### UI changes

Screenshot 2024-05-21 at 14 39 00

### CODE REVIEW

File: components/Header.tsx [hunk 1]
```diff
@@ -1,39 +1,70 @@
 'use client';
-import { usePathname } from 'next/navigation';
-import Link from 'next/link';
-import Image from 'next/image';
 import useMediaQuery from './hooks/useMediaQuery';
+import { useConversationContext } from '@/context/conversationContext';
+import { usePathname } from 'next/navigation';
+import { useEffect } from 'react';
 
 // Components
```
Comment (camelPhonso) [inline components/Header.tsx:7]:
```suggestion
```

while we're here, why not get rid of a needless comment? 🤷

Comment (camelPhonso) [reply components/Header.tsx:7]: I just learned this trick @nichgalzin - apparently you can just accept this suggestion and commit the change from here 🤯

File: components/layout/FullHeightComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    window.addEventListener('resize', handleResize);
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, []);
+
+  const style = {
+    height: height + 'px',
+    overflow: 'auto', // Add this style if you want the component to be scrollable
```
Comment (camelPhonso) [inline components/layout/FullHeightComponent.tsx:22]:
```suggestion
    overflow: 'auto',
```
Don't need that comment do we?

File: components/Header.tsx [hunk 3]
```diff
@@ -1,39 +1,70 @@
...
+import Image from 'next/image';
+import Link from 'next/link';
 import DesktopNav from './navigation/DesktopNav';
 import NavigationLinkContainer from './navigation/NavigationLinkContainer';
 import ProfileRouteIcon from './icons/navigation/ProfileRouteIcon';
+import { ConversationPartner } from './messaging/ConversationPartner';
```
Comment (camelPhonso) [inline components/Header.tsx:13]: This looks so much nicer now on the UI✨

**Review by camelPhonso — APPROVED** [review]
I've made some Inline Commit Suggestions because comments are not code and (mostly) because I wanted to try that new trick - but apart from those I'm happy for this to get shipped ⛵ 

Thank you for the detailed comments on your PR 🖤

File: components/Header.tsx (hunk 1 above)
Comment (nichgalzin) [reply components/Header.tsx:7]: This is actually a new convention we are going to try and adopt going forward to make it clear which components are imported from our own code and what's from outside libraries.

**Review by camelPhonso — APPROVED** [review]

**Pull request merged**